integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -500,29 +500,25 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
-@pytest.mark.parametrize('ansi_mode', ['nonAnsi', 'ansi'])
-@pytest.mark.parametrize('exp_type', ['bothLiterals', 'justZeroLiteral', 'noLiterals'])
-def test_div_by_zero(ansi_mode, exp_type):
-    if ansi_mode == 'ansi':
-        if is_before_spark_310():
-            pytest.xfail('https://github.com/apache/spark/pull/29882')
-        elif exp_type != 'bothLiterals':
-            pytest.xfail('https://github.com/NVIDIA/spark-rapids/issues/1464')
-
+@pytest.mark.parametrize('ansi_mode, expr', [
+    pytest.param('ansi', '1/0', marks=pytest.mark.xfail(condition=is_before_spark_310(),
+                                                        reason='https://github.com/apache/spark/pull/29882')),
+    pytest.param('ansi', 'a/0', marks=pytest.mark.xfail(condition=is_before_spark_310(),
+                                                        reason='https://github.com/apache/spark/pull/29882')),
+    pytest.param('ansi', 'a/b', marks=pytest.mark.xfail(condition=is_before_spark_310(),
+                                                        reason='https://github.com/NVIDIA/spark-rapids/issues/1464')),
```

Comment:
If this PR closes #1464 then why are we referencing it in new code here?

Reply:
> What's up with all the commits in this PR? Did you end up rebasing commits in your branch that are already committed to branch-0.4?

not knowingly or intentionally, I'll recreate the branch to get rid of it.